CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -501,15 +502,67 @@ function Test-VRouterAgentIntegration {
+            Write-Host "===> Running: Test-ICMPoMPLSoUDP"
+
+            $TestConfigurationTemp = $TestConfiguration.ShallowCopy()
+            $TestConfigurationTemp.DockerDriverConfiguration = $TestConfiguration.DockerDriverConfiguration.ShallowCopy()
+            $TestConfigurationTemp.ControllerIP = $Env:CONTROLLER_IP_UDP
+            $TestConfigurationTemp.DockerDriverConfiguration.AuthUrl = $Env:DOCKER_DRIVER_AUTH_URL_UDP
+
+            $ContrailUrl = $TestConfigurationTemp.ControllerIP + ":" + $TestConfigurationTemp.ControllerRestPort
+            $ContrailCredentials = $TestConfigurationTemp.DockerDriverConfiguration
+            $AuthToken = Get-AccessTokenFromKeystone -AuthUrl $ContrailCredentials.AuthUrl -TenantName $ContrailCredentials.TenantConfiguration.Name `
+                -Username $ContrailCredentials.Username -Password $ContrailCredentials.Password
+            $RouterIp1 = Invoke-Command -Session $Session1 -ScriptBlock {
+                return $((Get-NetIPAddress -InterfaceAlias $Using:TestConfigurationTemp.VHostName -AddressFamily IPv4).IpAddress)
+            }
+            $RouterIp2 = Invoke-Command -Session $Session2 -ScriptBlock {
+                return $((Get-NetIPAddress -InterfaceAlias $Using:TestConfigurationTemp.VHostName -AddressFamily IPv4).IpAddress)
+            }
+            $RouterUuid1 = Add-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterName $Session1.ComputerName -RouterIp RouterIp1
+            $RouterUuid2 = Add-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterName $Session2.ComputerName -RouterIp RouterIp2
+
+            Try {
+                Test-Ping -Session1 $Session1 -Session2 $Session2 -TestConfiguration $TestConfigurationTemp -Container1Name "container1" -Container2Name "container2"
+            } Finally {
+                Remove-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterUuid $RouterUuid1
+                Remove-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterUuid $RouterUuid2
+            }
+
+            $wasMplsUdpUsed = Invoke-Command -Session $Session1 -ScriptBlock {
+                $vrfstatsOutput = $(vrfstats --get 1)
+                $mplsUdpPktCount = [regex]::new("Udp Mpls Tunnels ([0-9]+)").Match($vrfstatsOutput[3]).Groups[1].Value
+                $mplsGrePktCount = [regex]::new("Gre Mpls Tunnels ([0-9]+)").Match($vrfstatsOutput[3]).Groups[1].Value
+                $vxlanPktCount = [regex]::new("Vxlan Tunnels ([0-9]+)").Match($vrfstatsOutput[3]).Groups[1].Value
+                if ($mplsUdpPktCount -gt 0 -and $mplsGrePktCount -eq 0 -and $vxlanPktCount -eq 0) {
+                    return $true
+                } else {
+                    return $false
+                }
+            }
+            if (!$wasMplsUdpUsed) {
+                throw "Containers pinged themselves correctly but used the wrong type of tunnel"
```

Comment:
Maybe it would be good to add pkt counters to exception. It could give user more information.